paymentsheet/src/main/java/com/stripe/android/link/account/DefaultLinkAccountManager.kt

```diff
@@ -449,21 +461,23 @@ internal class DefaultLinkAccountManager @Inject constructor(
         }
     }
 
-    private suspend fun LinkAccount?.fetchAccountStatus(shouldLookup: Boolean): AccountStatus =
-        /**
-         * If we already fetched an account, return its status, otherwise if a customer
-         * email was passed in (and the user hasn't logged out) lookup the account.
-         */
-        this?.accountStatus
-            ?: config.customerInfo.email
-                ?.takeIf { shouldLookup }
-                ?.let { customerEmail ->
-                    lookupConsumer(customerEmail).map {
-                        it?.accountStatus
-                    }.getOrElse {
-                        AccountStatus.Error
-                    }
-                } ?: AccountStatus.SignedOut
+    private suspend fun getAccountStatus(
+        linkAccount: LinkAccount?,
+        canLookupCustomerEmail: Boolean
+    ): AccountStatus {
```

Comment:
[non-blocking] I feel like this (and what we had before) is a bit confusing. What about something like:

- we've a `LinkAccount.getStatus(): AccountStatus`
- where we need the account status, we do:

`(linkAccount ?: lookup()).getStatus() ?: AccountStatus.Error`

`linkAccount ?: lookup()` can be a getOrFetchLinkAccount(canLookupCustomerEmail)` function, or similar. 

We can take a look at the approach on a different PR